### PR TITLE
Added Azure.Storage.DefenderCloud.MalwareScan

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Part of #2206

Added Azure.Storage.DefenderCloud.MalwareScan rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
@BernieWhite Lemme know and as well:

- Notes section about behaviour, does it look correct to you? I've done some testing and trying to interpret the documentation, but the documentation is a bit unclear about if it really creates it's own dedicated plan no matter what. Please check.
- I see we also have to create a general seperate rule for just Defender on resource level in order for this and #2207 to make the best sense?

File: src/PSRule.Rules.Azure/rules/Azure.Storage.Rule.ps1 [hunk 1]
```diff
@@ -88,6 +88,13 @@ Rule 'Azure.Storage.ContainerSoftDelete' -Ref 'AZR-000289' -Type 'Microsoft.Stor
     }
 }
 
+# Synopsis: Enable Malware Scanning in Microsoft Defender for Storage.
+Rule 'Azure.Storage.DefenderCloud.MalwareScan' -Ref 'AZR-000384' -Type 'Microsoft.Storage/storageAccounts' -Tag @{ release = 'Preview'; ruleSet = '2023_06'; 'Azure.WAF/pillar' = 'Security'; } -Labels @{ 'Azure.MCSB.v1/control' = 'DP-2', 'LT-1' } {
+    $malwareConfigured = @(GetSubResources -ResourceType 'Microsoft.Security/DefenderForStorageSettings' |
+        Where-Object { $_.properties.malwareScanning.onUpload.isEnabled -eq $True })
+    $Assert.GreaterOrEqual($malwareConfigured, '.', 1).Reason($LocalizedData.ResStorageMalwareScanning, $PSRule.TargetName)
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.Storage.Rule.ps1:95]: We want to find cases when the malware scanning has been disabled. Try the following instead.

```suggestion
    $malwareDisabled = @(GetSubResources -ResourceType 'Microsoft.Security/DefenderForStorageSettings' |
        Where-Object { $_.properties.malwareScanning.onUpload.isEnabled -eq $False })
    $Assert.Count($malwareDisabled, '.', 0).Reason($LocalizedData.ResStorageMalwareScanning, $PSRule.TargetName)
```

Additionally, I think we need to a pre-filter to skip the rule if `properties.publicNetworkAccess` is set to `Disabled` which is not supported as mentioned.

File: docs/en/rules/Azure.Storage.DefenderCloud.MalwareScan.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+Malware Scanning in Microsoft Defender for Storage can be enabled at the resource level. However, the general recommandation is to enable it at the subscription level and by doing so ensures all storage accounts in the subscription will be protected, including future ones. Defender for Storage settings on each storage account is inherited by the subscription level settings.
+
+It is also worth to mention that the resouce level enablement can be useful when:
+
+- Override subscription level settings to configure specific storage accounts with custom malware scanning settings that differ from the settings configured at the subscription level.
+
+## RECOMMENDATION
+
+Consider using Malware Scanning in Microsoft Defender for Storage.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Storage.DefenderCloud.MalwareScan.md:40]:
```suggestion
Consider enabling Malware Scanning using Microsoft Defender for Storage on the Storage Account.
Alternatively, enable Malware Scanning for all Storage Accounts within a subscription.
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Thanks @BenjaminEngeset. A few suggested changes. Separate rule is good.

Let me know if that doesn't answer your question.

**Comment by BenjaminEngeset** [comment]
Hi @BernieWhite I have done some adjustments and I'd like you to check them. As the rule is working now, only storage accounts with sub-resource set to disabled fails and else passes, but I guess that is what you had in mind, and rather use the subscription rules as the mains, right?

**Comment by BernieWhite** [comment]
Correct only storage accounts explicitly disabling the malware scan should fail.

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset . All good to merge.

**Pull request merged**